configs/swarm/docker-compose.yml

```diff
@@ -1,46 +1,60 @@
+# You should not edit this manually.
 
-version: '3.8'
 services:
+  image_provider:
+    image: hyperledger/iroha:dev
+    pull_policy: missing
+    command: echo ok
   irohad0:
+    depends_on:
+    - image_provider
     image: hyperledger/iroha:dev
+    pull_policy: never
     environment:
-      CHAIN_ID: 00000000-0000-0000-0000-000000000000
-      PUBLIC_KEY: ed012082528CCC8727333530C8F6F19F70C23882DEB1BF2BA3BE4A6654C7E8A91A7731
-      PRIVATE_KEY: 802640418A3712F4841FFE7A90B14E90BF76A6EF2A2546AC8DBBB1F442FFB8250426B082528CCC8727333530C8F6F19F70C23882DEB1BF2BA3BE4A6654C7E8A91A7731
+      CHAIN: 00000000-0000-0000-0000-000000000000
+      PUBLIC_KEY: ed012018CC218A5830112811C85287E43CEE37A81CBACEA72CFAB6A855E4A89BD60A91
+      PRIVATE_KEY: 802640268A94EC4A0FD1BEEC8ABD0A6CC184BCC5090698ADFF08758016D509F75FB6EC18CC218A5830112811C85287E43CEE37A81CBACEA72CFAB6A855E4A89BD60A91
       P2P_ADDRESS: 0.0.0.0:1337
       API_ADDRESS: 0.0.0.0:8080
-      GENESIS_PUBLIC_KEY: ed01204164BF554923ECE1FD412D241036D863A6AE430476C898248B8237D77534CFC4
+      GENESIS_PUBLIC_KEY: ed0120AC9417E94E01FC2C3120AC6D6CE51C1567FE80BFBDA7A11A431D8D81E88A50C8
+      SUMERAGI_TRUSTED_PEERS: '[{"address":"irohad3:1340","public_key":"ed01205A8F726921298226365765DA6A3DE40C8EFD650A60081951CFFCFA69255CEF60"},{"address":"irohad1:1338","public_key":"ed012064B29A7C9F90B223FE062D1BFFD794773C3385D3DE2E44FC935F7E5F97E37F6A"},{"address":"irohad2:1339","public_key":"ed0120BF4D7AB1A245F6432A2D8BD8C9D4F947FF750F9C35D9BD1F60E55A55DA50DDBE"}]'
       GENESIS_SIGNED_FILE: /tmp/genesis.signed.scale
-      SUMERAGI_TRUSTED_PEERS: '[{"address":"irohad1:1338","public_key":"ed012083C85E315776FD2DDC187ECB23E608F800B313A1D614B108078EC048D5013D2D"},{"address":"irohad2:1339","public_key":"ed0120A37B7B758C952FE9429E9E35D1D71E2D8BB9364EDD077B5027ABAAC798D3230E"},{"address":"irohad3:1340","public_key":"ed0120B23E14F659B91736AAB980B6ADDCE4B1DB8A138AB0267E049C082A744471714E"}]'
     ports:
     - 1337:1337
     - 8080:8080
     volumes:
     - ./:/config
     init: true
-    command: |-
-      /bin/sh -c "
-      kagami genesis sign /config/genesis.json --public-key $$GENESIS_PUBLIC_KEY --private-key 80264082B3BDE54AEBECA4146257DA0DE8D59D8E46D5FE34887DCD8072866792FCB3AD4164BF554923ECE1FD412D241036D863A6AE430476C898248B8237D77534CFC4 --out-file $$GENESIS_SIGNED_FILE &&
-      irohad --submit-genesis
-      "
     healthcheck:
       test: test $(curl -s http://127.0.0.1:8080/status/blocks) -gt 0
       interval: 2s
       timeout: 1s
       retries: 30
       start_period: 4s
+    command: |-
+      /bin/sh -c "
+        kagami genesis sign /config/genesis.json \
+          --public-key ed012018CC218A5830112811C85287E43CEE37A81CBACEA72CFAB6A855E4A89BD60A91 \
+          --private-key 802640268A94EC4A0FD1BEEC8ABD0A6CC184BCC5090698ADFF08758016D509F75FB6EC18CC218A5830112811C85287E43CEE37A81CBACEA72CFAB6A855E4A89BD60A91 \
+          --out-file /tmp/genesis.signed.scale && \
```

Comment:
```suggestion
          --out-file $GENESIS_SIGNED_FILE && \
```

Reply:
Apparently, Docker Compose doesn't recognize variables from the `environment` field, expecting them to be set in the host environment.